synfig-studio/src/gui/dialogs/dialog_setup.cpp

```diff
@@ -1446,6 +1467,28 @@ Dialog_Setup::on_brush_path_remove_clicked()
 	//! TODO if list size == 0: push warning to warning zone
 }
 
+void
+Dialog_Setup::on_plugin_path_change_clicked()
+{
+	String foldername = ResourceHelper::get_plugin_path();
+	//processes the path from get_plugin_path()
+	size_t pos = foldername.find('/');
+    while (pos != std::string::npos) {
+        foldername.replace(pos, 1, "\\");
+        pos = foldername.find('/', pos + 2);
+    }
+	#ifdef _WIN32
+    ShellExecuteA(NULL, "open", "explorer.exe", foldername.c_str(), NULL, SW_SHOWDEFAULT);
+	#elif __linux__
+    std::string command = "xdg-open " + foldername;
+    std::system(command.c_str());
+	#elif __APPLE__
+    std::string command = "open " + foldername;
+    std::system(command.c_str());
+	#else
+	return 0;
+	#endif
```

Comment:
1. there is an indentation issue here
2. you used `return 0` in a function that do not return anything (`void`)
3. it does not support OS that do not use UTF-8 encoding :(
4. Try to use `synfig::OS::launch_file_async()` ;)